ci/ci.sh

```
@@ -3,9 +3,17 @@ set -xe
 
 # Defaults (in case they are not set by the CI)
 F=${F:-production}
-DC=${DC:-dmd1}
 DIST=${DIST:-xenial}
 
+# Binary/package name deduced based on supplied DMD version
+case "$DMD" in
+    1.*    ) DC=dmd1 ;;
+    2.*.s* ) DC=dmd-transitional ;;
+    *      ) DC=dmd ;;
+esac
+
+apt install -y $DC=$DMD-$DIST
```

@leandro-lucarella-sociomantic how do you envision doing this correctly? (I can't `apt install` when script is executed and there are no env vars to check during `docker build` AFAICS)

---

Yeah, good point, we need some support in the beaver side. I will have a look.

---

For now, as a workaround, you can do the following (eventually I will integrate this into beaver, but we can give it a go first here):

```yml
install: beaver docker build --build-arg "DMD=$DMD" -f "Dockerfile${DIST:+.$DIST}" .
```

(this is basically what `beaver install` does plus passing the `DMD` argument to the build command).

Then in the `Dockerfile`:
```
ENV DMD=$DMD
```

Then you can use the `DMD` env variable from inside the build script.

---

Although with your current script I think the default would be broken, I think it might be better to default to something that makes sense (here the installed dmd1).

---

Build args reference: https://docs.docker.com/engine/reference/builder/#arg

---

Also, please create an issue in beaver about this (if there isn't one yet, there might be).

---

> Although with your current script I think the default would be broken

I actually wanted to raise the issue about current usage of defaults because it causes matrix builds to silently succeed (with default values) if something gets broken in env var passing. To me it is more important than minor convenience of running scripts manually.

Will try `--build-arg` approach tomorrow.

---

Mmm, good point. I just like/find convenient to be able to use the Dockerfile as is because I do it regularly. Maybe we can have some sort of utility script to define defaults for manual use or something.